go/vt/vtadmin/vtctldclient/proxy_test.go

```diff
@@ -34,16 +35,25 @@ type fakeVtctld struct {
 	vtctlservicepb.VtctlServer
 }
 
-func TestDial(t *testing.T) {
+func initVtctlServer() (net.Listener, *grpc.Server, error) {
 	listener, err := net.Listen("tcp", "127.0.0.1:0")
-	require.NoError(t, err)
-
-	defer listener.Close()
+	if err != nil {
+		return nil, nil, err
+	}
 
 	vtctld := &fakeVtctld{}
 	server := grpc.NewServer()
 	vtctlservicepb.RegisterVtctlServer(server, vtctld)
 
+	return listener, server, err
+}
+
+func TestDial(t *testing.T) {
```

Comment:
what you're seeing running just this test in isolation is super interesting, do you mind filing an issue and i can dig into it? i'm not really sure what's going, but it could be a "macs suck at local networking" issue, or something not completely correct in the code, but it's hard to say (and i don't think we should block this PR, which I still maintain is strictly an improvement over the current, uh, Situation)

Reply:
Done: https://github.com/vitessio/vitess/issues/9943

Thanks for taking a look. I'm super interested in what you find! I spent way too long looking at gRPC internals for this PR 😭